.github/workflows/cache-cleanup.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow cleans up any cache entries associated with a pull request once it has been closed.
+# This prevents us from having many refs/pull/PR_NUMBER/merge cache entries which will never be used.
+#
+# Note that this will affect both PRs being closed with and without being merged.
+
+name: Cleanup closed PR cache entries
+
+on:
+  pull_request:
+    types:
+      - closed
```

Comment:
I'm not sure if this fails on PRs submitted from forks because those shouldn't(?) have write permissions to the cache. But I'm not sure if the cache has different permission model than other action stuff.

Reply:
Possible alternative is to run on a schedule and fetch all closed PRs in the last day and trim based on that?

Reply:
Looks like PRs from forks do actually write to the cache.

Reply:
https://github.com/noir-lang/noir/actions/runs/5875108262/job/15930854582#step:15:5